### PR TITLE
Selectable tip amounts

### LINKED ISSUE #1371: Add pre-canned set of payments to Custom Banner
{ 1, 5, 10 } 
{ 5, 10, 20} 
{ 10, 20, 50 } 
{ 20, 50, 100 } 
{ 50, 100, 500 }

These are 5 sets of triplet values for L-M-H against the custom banner.
Allow any set per custom banner (per channel)

(Added by Albert): Please also dynamically load via cache the conversion rate.

### PR DESCRIPTION
**Pull request by dlipeles**
Closes #1371 

screen shot 2018-12-07 at 4 08 45 pm

### CODE REVIEW

**Comment by dlipeles** [comment]
Functionality complete, working now to add back-end validations and see if I can improve the alignment of the selectable options

File: app/javascript/packs/banner_editor.jsx [hunk 1]
```diff
@@ -421,6 +423,46 @@ export default class BannerEditor extends React.Component {
       }
   }
 
+  tipToOption(){
```
Comment (corymcdonald) [inline app/javascript/packs/banner_editor.jsx]: Think we could remove this option if we set a `donationAmountIndex` in the state? I think that might make more sense

**Comment by corymcdonald** [comment]
Can we update to the latest version of brave-ui?

**Comment by corymcdonald** [comment]
Can we ensure that when BAT moons that we don't have this problem 😉 

screen shot 2018-12-07 at 5 02 49 pm

**Comment by dlipeles** [comment]
WIP - Functionality and Validations are done , and was able to put in ev's request for right aligned values. 

Still want to improve some cosmetic pieces - 

- Use `|` as a separator instead of commas
- Replace inline styles with `styled-components`
- Use constants instead of literals in the validation

File: app/models/site_banner.rb [hunk 2]
```diff
@@ -27,10 +27,12 @@ class SiteBanner < ApplicationRecord
   #####################################################
 
   def donation_amounts_in_scope
+    amounts_literal = donation_amounts.join(',')
     return if errors.present? # Don't bother checking against donation amounts if donation amounts are nil
     errors.add(:base, "Must have #{NUMBER_OF_DONATION_AMOUNTS} donation amounts") if donation_amounts.count != NUMBER_OF_DONATION_AMOUNTS
     errors.add(:base, "A donation amount is zero or negative") if donation_amounts.select { |donation_amount| donation_amount <= 0}.count > 0
     errors.add(:base, "A donation amount is above a target threshold") if donation_amounts.select { |donation_amount| donation_amount >= MAX_DONATION_AMOUNT}.count > 0
+    errors.add(:base, "Must be an approved tip preset") if !(amounts_literal == '1,5,10' || amounts_literal == '5,10,20' || amounts_literal == '10,20,50' || amounts_literal == '20,50,100' || amounts_literal == '50,100,500')
```
Comment (corymcdonald) [inline app/models/site_banner.rb]:
```suggestion
    errors.add(:base, "Must be an approved tip preset") unless (['1,5,10', '5,10,20', '10,20,50', '20,50,100', '50,100,500'].include? amounts_literal)


```

File: app/models/site_banner.rb [hunk 3]
```diff
@@ -27,10 +27,12 @@ class SiteBanner < ApplicationRecord
   #####################################################
 
   def donation_amounts_in_scope
+    amounts_literal = donation_amounts.join(',')
     return if errors.present? # Don't bother checking against donation amounts if donation amounts are nil
     errors.add(:base, "Must have #{NUMBER_OF_DONATION_AMOUNTS} donation amounts") if donation_amounts.count != NUMBER_OF_DONATION_AMOUNTS
     errors.add(:base, "A donation amount is zero or negative") if donation_amounts.select { |donation_amount| donation_amount <= 0}.count > 0
     errors.add(:base, "A donation amount is above a target threshold") if donation_amounts.select { |donation_amount| donation_amount >= MAX_DONATION_AMOUNT}.count > 0
```
Comment (corymcdonald) [inline app/models/site_banner.rb]: I think the above lines aren't needed if we're doing validation below

**Comment by dlipeles** [comment]
This is good enough to merge, but expect a code cleanup and refactor in #1452 
- Remove `document.getElementByID` calls
- Clean up handler and tip to Index method
- Right align BAT values in the dropdown

File: app/models/site_banner.rb [hunk 4]
```diff
@@ -28,6 +29,7 @@ def donation_amounts_in_scope
     errors.add(:base, "Must have #{NUMBER_OF_DONATION_AMOUNTS} donation amounts") if donation_amounts.count != NUMBER_OF_DONATION_AMOUNTS
     errors.add(:base, "A donation amount is zero or negative") if donation_amounts.select { |donation_amount| donation_amount <= 0}.count > 0
```
Comment (corymcdonald) [inline app/models/site_banner.rb]: We're already checking this by validating the approved tip presets right? Can we remove this line?

File: app/models/site_banner.rb [hunk 5]
```diff
@@ -28,6 +29,7 @@ def donation_amounts_in_scope
...
     errors.add(:base, "A donation amount is above a target threshold") if donation_amounts.select { |donation_amount| donation_amount >= MAX_DONATION_AMOUNT}.count > 0
```
Comment (corymcdonald) [inline app/models/site_banner.rb]: >We're already checking this by validating the approved tip presets right? Can we remove this line?
Same thing here

**Pull request merged**